apps/prairielearn/assets/stylesheets/pageLayout.css

```
@@ -198,6 +127,164 @@ body:has(div.app-container) {
+  .app-container.side-nav-enabled.collapsed .side-nav-link .suffix {
+    display: none;
+  }
+
+  /* When the side nav is collapsed, hide the expanded icon */
+  .app-container.side-nav-enabled.collapsed .side-nav-toggler-icon.expanded {
+    display: none;
+  }
+
+  /* When the side nav is expanded, hide the collapsed icon */
+  .side-nav-toggler-icon.collapsed {
+    display: none;
+  }
+
+  /* When the side nav is collapsed, show the collapsed icon */
+  .app-container.side-nav-enabled.collapsed .side-nav-toggler-icon.collapsed {
+    display: block;
+  }
+
+  .app-container.side-nav-enabled.collapsed #side-nav-toggler {
+    padding: 0.5rem 1rem;
+  }
+}
+
+@media (max-width: 768px) {
+  .app-container.side-nav-enabled {
+    display: block;
+  }
+
+  .app-container.side-nav-enabled #side-nav-toggler {
+    display: none;
+  }
+
+  .app-container.side-nav-enabled.mobile-collapsed .app-side-nav {
+    transform: translateX(-100%);
+  }
+
+  .app-container.side-nav-enabled:not(.mobile-collapsed) .app-side-nav {
+    transform: translateX(0);
+  }
+
+  .app-container.side-nav-enabled .app-side-nav {
+    display: block;
+    position: absolute;
+    padding-top: 1rem;
+    top: 56px;
+    left: 0;
+    width: 100%;
+    z-index: 999;
+    transition: transform 0.3s ease-in-out;
+    transform: translateX(-100%); /* Start off-screen to the left */
+    will-change: transform; /* Performance optimization */
+    min-height: 100vh;
+  }
+
+  .app-container.side-nav-enabled #course-nav {
+    position: absolute;
+    top: 48px;
+    left: 0;
+    width: 100%;
+    height: fit-content;
+    min-height: 100vh;
+    z-index: 999;
+    background-color: rgba(var(--bs-dark-rgb));
+    padding: 1rem 0.5rem;
+    transition: transform 0.3s ease-in-out;
+    transition-delay: 0;
+    will-change: transform; /* Performance optimization */
+    transform: translateX(100%); /* Start off-screen to the right */
+  }
+
+  .app-container.side-nav-enabled #course-nav.show {
+    transform: translateX(0);
+  }
+
+  .app-container.side-nav-enabled #course-nav:not(.show) {
+    transform: translateX(100%);
+  }
+
+  /* Prevent content under the course/side nav from being visible when
+  either nav is open. */
+  .app-container.side-nav-enabled:has(#course-nav.show) .app-main,
+  .app-container.side-nav-enabled:not(.mobile-collapsed) .app-main {
+    height: 100%;
+    overflow-y: hidden;
+  }
+
+  .app-container.side-nav-enabled:has(#course-nav.show),
+  .app-container.side-nav-enabled:not(.mobile-collapsed) {
+    overflow-x: hidden;
+  }
+
+  /* If the side nav is collapsed on a wider viewports,
+  *  tooltips appear above the side nav link icons when hovered. 
+  *  
+  *  If the user collapses the side nav on a wide viewport, then 
+  *  resizes their window to a narrow viewport (less than 768px), 
+  *  the tooltips should be hidden, since side nav link text
+  *  is displayed on narrow viewports. */
```

Formatting/description nits:

```suggestion
  /**
   * On wider viewports, when the side nav is collapsed,
   * tooltips appear next to the side nav link icons when hovered. 
   *
   * This rule hides those tooltips on narrow viewports.
   */
```